MakeHist/HTLT.C

```diff
@@ -98,4 +128,102 @@ void HTLT(){
 
     leg_HTLT->Draw();
 
+    c1->SaveAs("./hists/"+prefix+"_HTLT.png");
+
+
+    //make bkg for siginificance plot
+    HTLT_DB->Add(HTLT_ST);
+    HTLT_DB->Add(HTLT_TT);
+    HTLT_DB->Add(HTLT_DY);
+
+    TH1F * HTLT_DB_copy = (TH1F*) HTLT_DB->Clone("DB_copy");
+
+    TH1F * HTLT_200_copy = (TH1F*) HTLT_200->Clone("200_copy");
+
+    //std::cout << "ComputeIntegral " << HTLT_DB->ComputeIntegral() << "\n";
+    float previousBin;
+    float currentBin;
+    int nBins = HTLT_DB->GetNbinsX();
+
+    //std::cout << "right cut \n";
+    for(int i=1;i<nBins+2;i++){
+
+        previousBin = HTLT_DB->GetBinContent(i-1);
+        currentBin = HTLT_DB->GetBinContent(i);
+
+        HTLT_DB->SetBinContent(i, currentBin+ previousBin);
+        //std::cout << HTLT_DB->GetBinContent(i) << "\n";
+
+        previousBin = HTLT_200->GetBinContent(i-1);
+        currentBin = HTLT_200->GetBinContent(i);
+
+        HTLT_200->SetBinContent(i, currentBin+ previousBin);
+        //std::cout << HTLT_200->GetBinContent(i) << "\n";
+
+
+
+    }
+
+
+
+    std::cout << "left cut \n";
+    for(int i=nBins;i>-1;i--){
+
+        previousBin = HTLT_DB_copy->GetBinContent(i+1);
+        currentBin = HTLT_DB_copy->GetBinContent(i);
+
+        HTLT_DB_copy->SetBinContent(i, currentBin+ previousBin);
+        //std::cout << HTLT_DB_copy->GetBinContent(i) << "\n";
+
+        previousBin = HTLT_200_copy->GetBinContent(i+1);
+        currentBin = HTLT_200_copy->GetBinContent(i);
+
+        HTLT_200_copy->SetBinContent(i, currentBin+ previousBin);
+        //std::cout << HTLT_200_copy->GetBinContent(i) << "\n";
+
+
+    }
+
+      //HTLT_200->Divide(HTLT_DB);
+    HTLT_200->Draw();
+    c1->SaveAs("./hists/"+prefix+"_HTLT_right_cut.png");
+
+    HTLT_DB->Draw();
+    c1->SaveAs("./hists/"+prefix+"_HTLT_DB_right_cut.png");
+
+
+    HTLT_200->Divide(HTLT_DB);
+    HTLT_200->Draw();
+    c1->SaveAs("./hists/"+prefix+"_HTLT_right_cut_s_over_b.png");
+    HTLT_200->Multiply(HTLT_DB);
+
+    HTLT_DB->Add(HTLT_200);
+    HTLT_200->Multiply(HTLT_200);
+    HTLT_200->Divide(HTLT_DB);
+    HTLT_200->Draw();
```

Comment:
Why go this weird way to S^2/(S+B) and not simply do
`TH1F *significance_left=(TH1F*)HTLT_DB->Clone("significance");
TH1F *significance_right=(TH1F*)HTLT_TB->Clone("significance"); //TB is for the total background hist
const int maxBin=HTLT_DB->GetNbinsX();
for(int i=0; i<maxBin; ++i){
  const float S_left=HTLT_200->Integral(i,-1);
  const float S_right=HTLT_200->Integral(0,maxBin-i);
  const float B_left=HTLT_TB->Integral(i,-1);
  const float B_right=HTLT_TB->Integral(0,maxBin-i);
  significance_left->SetBinContent(i,S_left/sqrt(S_left+B_left));
  significance_right->SetBinContent(maxBin-i,S_right/sqrt(S_right+B_right));
}`
?

Reply:
It appeals to my numpy-vector operation's sensibilities, but ill change it.

Reply:
Technically this way also doesn't preserve/compute errors. I could set them manually.

Reply:
Use IntegralAndError to get the uncertainties, dump the 4 inputs into an uncertainty function, SetBinError as well, then.